airflow/contrib/hooks/aws_hook.py

```diff
@@ -132,7 +134,7 @@ def _get_credentials(self, region_name):
                     role_arn = "arn:aws:iam::{}:role/{}" \
                         .format(aws_account_id, aws_iam_role)
 
-                if role_arn is not None:
+                if role_arn is not None and aws_session_token is None:
```

Comment:
I'm not sure this condition makes sense. What if I want to use key+secret+token to assume a role?

Reply:
Not sure if you want to obtain a token if you already have one ?

Reply:
If `role_arn` is provided in the config that role should be assumed. Security Token being present or not is an orthogonal concept to assuming a role.